Framework/Algorithms/test/DiscusMultipleScatteringCorrectionTest.h

```diff
@@ -95,28 +95,32 @@ class DiscusMultipleScatteringCorrectionTest : public CxxTest::TestSuite {
     Mantid::API::Workspace_sptr wsPtr = output->getItem("Scatter_2");
     auto doubleScatterResult = std::dynamic_pointer_cast<Mantid::API::MatrixWorkspace>(wsPtr);
 
-    // validate that the max scatter angle is ~94 degrees
+    // validate that the max scatter angle is ~115 degrees (peak is at 114 but slight tail)
     for (size_t i = 0; i < NTHETA; i++)
-      if (doubleScatterResult->spectrumInfo().twoTheta(i) > M_PI * 94.0 / 180.0)
+      if (doubleScatterResult->spectrumInfo().twoTheta(i) > M_PI * 115.0 / 180.0)
         TS_ASSERT_EQUALS(doubleScatterResult->y(i)[0], 0.);
 
-    // crude check on peak positions at theta=0 and ~94 degrees
+    // crude check on peak positions at theta=0 and ~114 degrees
     double sum = 0.;
     for (size_t i = 0; i < NTHETA; i++)
       sum += doubleScatterResult->y(i)[0];
     double avgY = sum / NTHETA;
     std::vector<size_t> peakPos;
     int PEAKSPACING = NTHETA / 10;
-    int lastPeakFound = -PEAKSPACING;
-    for (size_t i = 0; i < NTHETA; i++)
-      if ((doubleScatterResult->y(i)[0] > 3 * avgY) && (static_cast<int>(i - lastPeakFound) >= PEAKSPACING)) {
-        peakPos.push_back(i);
-        lastPeakFound = static_cast<int>(i);
+    for (int i = 0; i < NTHETA; i++) {
+      bool maxInWindow = true;
+      for (int j = std::max(0, i - PEAKSPACING); j <= std::min(NTHETA - 1, i + PEAKSPACING); j++) {
+        if (doubleScatterResult->y(j)[0] > doubleScatterResult->y(i)[0])
+          maxInWindow = false;
       }
+      if ((doubleScatterResult->y(i)[0] > 3 * avgY) && maxInWindow)
+        peakPos.push_back(i);
+    }
     TS_ASSERT_EQUALS(peakPos.size(), 2);
     if (peakPos.size() > 0) {
       TS_ASSERT_EQUALS(peakPos.front(), 0);
-      TS_ASSERT((static_cast<double>(peakPos.back()) * 0.2 > 93) && (static_cast<double>(peakPos.back()) * 0.2 < 94));
+      TS_ASSERT((static_cast<double>(peakPos.back()) * 0.2 >= 114) &&
```

Comment:
Nice addition to the test, where does the factor of 0.2 come from, might be worth a comment?

Reply:
The factor converts a workspace index to a two theta value. It's basically 180 degrees divided by NTHETA so perhaps I should use that formula rather than hardcoding 0.2. It someone changes NTHETA the current code will break